wurst/event/DamageModify.wurst

```diff
@@ -0,0 +1,327 @@
+/**
+	DamageModify
+	by Trokkin
+
+	API:
+
+	// Pretty much '(Damage d) -> ...'
+	interface ForDamageCallback
+		run(Damage d)
+
+	// these are called before the damage is dealt
+	// can be called multiple amount of times or not called at all
+	addDamageModifier(ForDamageCallback cb)
+	removeDamageModifier(ForDamageCallback cb)
+	
+	// these are called after the damage is dealt
+	// are called strictly once per damage instance
+	addDamageListener(ForDamageCallback cb)
+	removeDamageListener(ForDamageCallback cb)
+
+	// a short for 'new Damage(this, target, amount, isSpell, id)..modify()..deal()'
+	unit.damage(unit target, real amount, OPTIONAL bool isSpell = true, int id = 0)
+
+	// a short for '= new Damage(this, target, amount, isSpell, id)..saveInstance()..modify()..deal()'
+	unit.returnDamage(unit target, real amount, bool isSpell, int id) -> Damage
+
+	class Damage
+		
+		// Basic damage event info
+		unit source
+		unit target
+		bool spell
+		real initial
+
+		// .id is used for any unique iterations
+		integer id
+		// .amount is used to manipulate damage amount
+		// like, addDamageModifier(d -> d.amount += d.initial * 1.00) adds +100% multiplier onto damage
+		real amount
+
+		// if .saveTarget was called then .target will survive from this damage no matter what
+		saveTarget()
+
+		// after damage has happened the Damage object is destroyed if saveInstance() wasn't called
+		saveInstance()
+		// releaseInstance is used to destroy Damage objects. Use this instead of 'destroy'!
+		releaseInstance()
+
+		// Some obvious wrappers
+		addAmount(real amount)
```

Comment:
Can you give an example of when this would be valuable, given that your `addDamageModifier` above is already quite succinct?

Reply:
That's what frotty has requested
Like, addDamageModifier() d -> d.addAmount(100500)
Probably just in case dear user wouldn't want to access .amount field directly

Reply:
That's what frotty has requested
Like, addDamageModifier() d -> d.addAmount(100500)
Probably just in case dear user wouldn't want to access .amount field directly